src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/events/ManageExternalAppointmentsService.kt

```diff
@@ -169,34 +173,32 @@ class ManageExternalAppointmentsService(
   }
 
   private fun Collection<AppointmentSearchResult>.findMatchingAppointments(appointment: PrisonAppointment): List<AppointmentSearchResult> =
-    filter {
-      appointment.startTime == LocalTime.parse(it.startTime) && appointment.endTime == LocalTime.parse(it.endTime)
+    filter { searchResult ->
+      searchResult.isTimesAreTheSame(appointment) && searchResult.isAppointmentType(supportedAppointmentTypes.typeOf(appointment)) && searchResult.isCancelled.not()
     }.ifEmpty {
       emptyList<AppointmentSearchResult>()
         .also { log.info("EXTERNAL APPOINTMENTS: no matching appointments found in A&A for prison appointment ${appointment.prisonAppointmentId}") }
     }
 
   private fun Collection<AppointmentSearchResult>.findMatchingAppointments(bha: BookingHistoryAppointment): List<AppointmentSearchResult> =
-    filter {
-      bha.startTime == LocalTime.parse(it.startTime) && bha.endTime == LocalTime.parse(it.endTime)
+    filter { searchResult ->
+      searchResult.isTimesAreTheSame(bha) && searchResult.isAppointmentType(supportedAppointmentTypes.typeOf(bha)) && searchResult.isCancelled.not()
```

Comment:
@smcveigh941 note the check here of cancellations in A&A, this should help with the other stuff for the daily schedule.

Will still need a further in A&A db to not delete cancelled appointments.